File: src/docs/Pages/BoxShadow.fs
module Docs.Pages.BoxShadow

open Feliz
open Elmish
open Feliz.UseElmish
open Feliz.DaisyUI
open Feliz.Tailwind
open Feliz.Tailwind.Operators
open Docs.SharedView

let boxShadowStyles =
    [ boxShadow.shadow2Xs
      boxShadow.shadowXs
      boxShadow.shadowSm
      boxShadow.shadowMd
      boxShadow.shadowLg
      boxShadow.shadowXl
      boxShadow.shadow2Xl
      boxShadow.shadowInner
      boxShadow.shadowNone ]

let str prop =
    if prop = boxShadow.shadow2Xs then
        "boxShadow.shadow2Xs"
    elif prop = boxShadow.shadowXs then
        "boxShadow.shadowXs"
    elif prop = boxShadow.shadowSm then
        "boxShadow.shadowSm"
    elif prop = boxShadow.shadowMd then
        "boxShadow.shadowMd"
    elif prop = boxShadow.shadowLg then
        "boxShadow.shadowLg"
    elif prop = boxShadow.shadowXl then
        "boxShadow.shadowXl"
    elif prop = boxShadow.shadow2Xl then
        "boxShadow.shadow2Xl"
    elif prop = boxShadow.shadowInner then
        "boxShadow.shadowInner"
    elif prop = boxShadow.shadowNone then
        "boxShadow.shadowNone"
    else
        failwith "Unknown BoxShadow property"
let renderBoxShadow boxShadow =
    let example =
        Daisy.button.button
            [   button.outline
                button.primary
                button.lg
                boxShadow
                prop.text (str boxShadow) ]

    let code =
        $"""Daisy.button.button
                [   button.outline
                    button.primary
                    button.lg
                    {str boxShadow}
                    prop.text "This is {str boxShadow}" ]"""

    let title = Html.text "Use of borderRadius is rather straightforward."
    codedView title code example

[<ReactComponent>]
let BoxShadowView () =
    React.fragment [
        for boxShadow in boxShadowStyles do
            renderBoxShadow boxShadow
    ]

File: src/docs/output/Pages/BoxShadow.js
import { ofArray } from "../fable_modules/fable-library-js.4.24.0/List.js";
import { createObj, equals } from "../fable_modules/fable-library-js.4.24.0/Util.js";
import { createElement } from "react";
import React from "react";
import * as react from "react";
import { Helpers_combineClasses } from "../fable_modules/Feliz.DaisyUI.5.0.0/DaisyUI.fs.js";
import { codedView } from "../SharedView.js";
import { map, delay, toList } from "../fable_modules/fable-library-js.4.24.0/Seq.js";

export const boxShadowStyles = ofArray([["className", "shadow-2xs"], ["className", "shadow-xs"], ["className", "shadow-sm"], ["className", "shadow-md"], ["className", "shadow-lg"], ["className", "shadow-xl"], ["className", "shadow-2xl"], ["className", "shadow-inner"], ["className", "shadow-none"]]);

export function str(prop) {
    if (equals(prop, ["className", "shadow-2xs"])) {
        return "boxShadow.shadow2Xs";
    }
    else if (equals(prop, ["className", "shadow-xs"])) {
        return "boxShadow.shadowXs";
    }
    else if (equals(prop, ["className", "shadow-sm"])) {
        return "boxShadow.shadowSm";
    }
    else if (equals(prop, ["className", "shadow-md"])) {
        return "boxShadow.shadowMd";
    }
    else if (equals(prop, ["className", "shadow-lg"])) {
        return "boxShadow.shadowLg";
    }
    else if (equals(prop, ["className", "shadow-xl"])) {
        return "boxShadow.shadowXl";
    }
    else if (equals(prop, ["className", "shadow-2xl"])) {
        return "boxShadow.shadow2Xl";
    }
    else if (equals(prop, ["className", "shadow-inner"])) {
        return "boxShadow.shadowInner";
    }
    else if (equals(prop, ["className", "shadow-none"])) {
        return "boxShadow.shadowNone";
    }
    else {
        throw new Error("Unknown BoxShadow property");
    }
}

export function renderBoxShadow(boxShadow) {
    const example = createElement("button", createObj(Helpers_combineClasses("btn", ofArray([["className", "btn-outline"], ["className", "btn-primary"], ["className", "btn-lg"], boxShadow, ["children", str(boxShadow)]]))));
    const code = `Daisy.button.button
                [   button.outline
                    button.primary
                    button.lg
                    ${str(boxShadow)}
                    prop.text "This is ${str(boxShadow)}" ]`;
    const title = "Use of borderRadius is rather straightforward.";
    return codedView(title, code, example);
}

export function BoxShadowView() {
    const xs = toList(delay(() => map(renderBoxShadow, boxShadowStyles)));
    return react.createElement(react.Fragment, {}, ...xs);
}

